Label maker: panic in wrapText (slice bounds out of range) returns HTTP 500 for many asset labels

### First Check

- [x] This is not a feature request
- [x] I added a very descriptive title to this issue.
- [x] I used the GitHub search to find a similar issue and didn't find it.
- [x] I searched the documentation, with the integrated search.
- [x] I already read the docs and didn't find an answer.
- [x] I can replicate the issue inside the Demo install.

### Homebox Version

v0.25.0 (also present in v0.26.2 and on main — pkgs/labelmaker/labelmaker.go is byte-identical between v0.25.0 and v0.26.2, and wrapText is unchanged on main)

### What is the issue you are experiencing?

`GET /api/v1/labelmaker/asset/{assetId}` returns HTTP 500 for many items. The label image never renders, so QR/asset labels are broken for those items. In my install **99 of 347 items** (~29%) fail; the rest render fine.

The server logs a panic in `wrapText`:

```
panic: runtime error: slice bounds out of range [54:53]

-> github.com/sysadminsmedia/homebox/backend/pkgs/labelmaker.wrapText
->   /go/src/app/pkgs/labelmaker/labelmaker.go:111

   github.com/sysadminsmedia/homebox/backend/pkgs/labelmaker.GenerateLabel
     /go/src/app/pkgs/labelmaker/labelmaker.go:255
   github.com/sysadminsmedia/homebox/backend/app/api/handlers/v1.generateOrPrint
     /go/src/app/app/api/handlers/v1/v1_ctrl_labelmaker.go:32
   main.(*app).mountRoutes.func1.(*V1Controller).HandleGetAssetLabel.82
     /go/src/app/app/api/handlers/v1/v1_ctrl_labelmaker.go:134
```

The panic is recovered by chi's `Recoverer`, so the process stays up and the request just 500s — no crash, no restart. Observed index overruns across different items include `[54:53]`, `[56:55]`, `[74:73]`, `[80:79]`.

### Cause

`wrapText` tracks consumed input with a `processedChars` counter and then slices the **original** string with it:

https://github.com/sysadminsmedia/homebox/blob/v0.25.0/backend/pkgs/labelmaker/labelmaker.go#L96-L148

The counter is reconstructed from the wrapped output rather than measured against the source, so the two drift apart:

1. Every emitted line advances the counter by `len(currentLine) + 1`, assuming exactly one separator byte follows it. The **last** line of each paragraph has no trailing separator, so each paragraph contributes a spurious `+1`.
2. `strings.Fields(line)` collapses runs of whitespace, so any multi-space run in the source is counted as a single space.

Once `currentHeight > maxHeight`, the three return paths slice `text` with that drifted counter:

- L111 — `text[processedChars:]`
- L130 / L142 — `text[processedChars-len(currentLine)-1:]`

None of them is bounds-checked, so when the counter has drifted past `len(text)` the slice panics. The drift grows with the number of emitted lines, which is why the overrun is not a fixed offset.

Note L111 is only reachable when `maxHeight != -1`. The only bounded call is the description wrap at L255, with `maxHeight = params.QrSize - titleHeight` — so an item panics when its body wraps to enough lines to hit that height cutoff.

For asset labels the body is built at [`v1_ctrl_labelmaker.go#L127-L131`](https://github.com/sysadminsmedia/homebox/blob/v0.25.0/backend/app/api/handlers/v1/v1_ctrl_labelmaker.go#L127-L131) as `item.Name + "\nLocation: " + location.Name` — the item's own `description` field is not involved. 32 of my 99 failing items have an empty description.

### How can the maintainer reproduce the issue?

With **default** labelmaker config (`width=526`, `height=200`, `padding=32`, `font_size=32`):

1. Create a location named `3D Printing Area`
2. Create an item in it named `WhamBam Flex Plate System`, leaving the description empty
3. `GET /api/v1/labelmaker/asset/{assetId}` for that item, or open its label in the UI

Result: HTTP 500, with the `wrapText` panic above (`[54:53]` for this exact pair).

This yields a body of `"WhamBam Flex Plate System\nLocation: 3D Printing Area"` (52 bytes). Other confirmed pairs from my install, same shape:

| Item name | Location |
|---|---|
| `Wanhao Duplicator i3 Plus` | `3D Printing Area` |
| `Guess the Princess (game)` | `CCE Storage Unit` |
| `BIGTREETECH Eddy Duo Probe` | `3D Printing Area` |

Because the trigger is where the wrap lands relative to the height cutoff, it is not predictable from string length alone — in my data, working and failing items overlap heavily in body length (failing 48–84 bytes, working 26–99).



### Suggested fix

Clamp the slice bounds at all three return sites, e.g.:

```go
if currentHeight > maxHeight {
    cut := processedChars - len(currentLine) - 1
    if cut < 0 {
        cut = 0
    }
    if cut > len(text) {
        cut = len(text)
    }
    return wrappedLines[:len(wrappedLines)-1], text[cut:]
}
```

That stops the panic, but the returned remainder can still be misaligned with the source (the drift is still there — it just no longer panics). A more complete fix would track the byte offset into `text` directly while scanning, instead of summing `len(currentLine)+1` per emitted line, so the overflow remainder is exact.

### Deployment

Docker (Linux)

### Database Type

SQLite

### OS Architechture

x86_64 (AMD, Intel)

### Deployment Details

Kubernetes (Docker Linux images), `sysadminsmedia/homebox:0.25.0`, single replica, SQLite on a PVC.

Label maker config is all upstream defaults: `width=526`, `height=200`, `padding=32`, `font_size=32`.

--- a/backend/pkgs/labelmaker/labelmaker.go
+++ b/backend/pkgs/labelmaker/labelmaker.go
@@ -109,7 +109,7 @@ func wrapText(text string, face font.Face, maxWidth int, maxHeight int, lineHeig
 			if !unlimitedHeight {
 				currentHeight += lineHeight
 				if currentHeight > maxHeight {
-					return wrappedLines[:len(wrappedLines)-1], text[processedChars:]
+					return wrappedLines[:len(wrappedLines)-1], text[min(processedChars, len(text)):]
 				}
 			}
 			continue

--- a/backend/pkgs/labelmaker/labelmaker_test.go
+++ b/backend/pkgs/labelmaker/labelmaker_test.go
@@ -1,12 +1,14 @@
 package labelmaker
 
 import (
+	"image"
 	"os"
 	"path/filepath"
 	"runtime"
 	"strings"
 	"testing"
 
+	"github.com/golang/freetype/truetype"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"github.com/sysadminsmedia/homebox/backend/internal/sys/config"
@@ -255,4 +257,24 @@ func TestPrintLabel_ArgumentInjection(t *testing.T) {
 	require.Len(t, got, 2, "argv must not be split on whitespace inside the title")
 	assert.Equal(t, maliciousTitle, got[0])
 	assert.True(t, strings.HasSuffix(got[1], ".png"), "second arg should be the generated label file, got %q", got[1])
+}
+
+func TestWrapText_TrailingEmptyLineAtHeightLimit(t *testing.T) {
+	regularFont, err := loadFont(nil, FontTypeRegular)
+	require.NoError(t, err)
+
+	face := truetype.NewFace(regularFont, &truetype.Options{
+		Size: 12,
+		DPI:  72,
+	})
+	tmpImg := image.NewRGBA(image.Rect(0, 0, 1, 1))
+	ctx := createContext(regularFont, 12, tmpImg, 72)
+
+	// The height limit is crossed on the trailing empty line, where
+	// processedChars has already advanced past len(text).
+	text := "Widget A\n"
+	lines, remaining := wrapText(text, face, 200, 12, 12, ctx)
+
+	assert.Equal(t, []string{"Widget A"}, lines)
+	assert.Empty(t, remaining)
 }

